public/app/features/dashboard/containers/NewDashboardWithDS.tsx

```diff
@@ -17,19 +22,24 @@ export default function NewDashboardWithDS(props: GrafanaRouteComponentProps<{ d
       return;
     }
 
-    const newDashboard = getNewDashboardModelData();
-    const { dashboard } = newDashboard;
-    dashboard.panels[0] = {
-      ...dashboard.panels[0],
-      datasource: {
-        uid: ds.uid,
-        type: ds.type,
-      },
-    };
-
-    setDashboardToFetchFromLocalStorage(newDashboard);
+    if (!config.featureToggles.emptyDashboardPage) {
+      const newDashboard = getNewDashboardModelData();
+      const { dashboard } = newDashboard;
+      dashboard.panels[0] = {
+        ...dashboard.panels[0],
+        datasource: {
+          uid: ds.uid,
+          type: ds.type,
+        },
+      };
+
+      setDashboardToFetchFromLocalStorage(newDashboard);
```

Comment:
the emptyDashboardPage FF is enabled by default now, are you sure we don't need `setDashboardToFetchFromLocalStorage(newDashboard);` in the case where it is enabled too? I don't know why we have it in the first place, that's why I'm asking

Reply:
Pretty sure. In this case it (was) used to set the initial dashboard that includes the panel with the pre-set datasource, but that behavior is handled with setInitialDatasource now